ksqldb-engine/src/main/java/io/confluent/ksql/analyzer/Analyzer.java

```diff
@@ -243,27 +218,6 @@ private KeyFormat buildKeyFormat() {
               .getKeyFormat());
     }
 
-    private void setSerdeOptions(final Sink sink) {
-      final List<ColumnName> columnNames = getColumnNames();
-
-      final Format valueFormat = getValueFormat(sink);
-
-      final Set<SerdeOption> serdeOptions = serdeOptionsSupplier.build(
-          columnNames,
-          valueFormat,
-          sink.getProperties().getWrapSingleValues(),
-          defaultSerdeOptions
-      );
-
-      analysis.setSerdeOptions(serdeOptions);
-    }
```

Comment:
To set the serde options we need to know the number of columns, which requires any `*` to be resolved. Hence this code has now moved to later `LogicalPlanner` once any `*` has been resolved.